### PR TITLE
feat: added ability to delete files with github client

### PR DESCRIPTION
**Pull request by logan-anderson**
added ability to delete files with github client

### CODE REVIEW

File: packages/react-tinacms-github/src/github-client/github-client.ts [hunk 1]
```diff
@@ -311,6 +315,30 @@ export class GithubClient {
...
+  async delete(
+    path: string,
+    fileContents: string,
+    commitMessage: string = 'Update from TinaCMS',
+    encoded: boolean = false
+  ) {
+    return this.githubFileApi(
+      path,
+      fileContents,
+      commitMessage,
```
Comment (ncphillips) [inline packages/react-tinacms-github/src/github-client/github-client.ts:336]: Maybe we could have a more informative default commit message like:

```
`Deleted ${path} using TinaCMS`
```

Comment (logan-anderson) [reply packages/react-tinacms-github/src/github-client/github-client.ts:336]: :+1:

Comment (logan-anderson) [reply packages/react-tinacms-github/src/github-client/github-client.ts:336]: @ncphillips  Fixed

File: packages/react-tinacms-github/src/github-client/github-client.ts [hunk 2]
```diff
@@ -311,6 +315,30 @@ export class GithubClient {
...
+
+  async delete(
+    path: string,
+    fileContents: string,
+    commitMessage: string = `Deleted ${path} using TinaCMS`,
+    encoded: boolean = false
+  ) {
+    return this.githubFileApi(
+      path,
+      fileContents,
```
Comment (ncphillips) [inline packages/react-tinacms-github/src/github-client/github-client.ts:335]: Are the file contents really needed for deleting?

Comment (logan-anderson) [reply packages/react-tinacms-github/src/github-client/github-client.ts:335]: nope

Comment (ncphillips) [reply packages/react-tinacms-github/src/github-client/github-client.ts:335]: Right maybe remove `fileContents` and `encoded` from this function's API:

```ts
async delete(
  path: string
  commitMessage: string = `Deleted ${path} using TinaCMS`,
) {
```

Comment (logan-anderson) [reply packages/react-tinacms-github/src/github-client/github-client.ts:335]: Not sure if this is the best of doing it. what do you think.

```ts
async delete(
    path: string,
    commitMessage: string = `Deleted ${path} using TinaCMS`,
    encoded: boolean = false
  ) {
    return this.githubFileApi(path, '', commitMessage, encoded, 'DELETE')
  }

the contents are just ignored by the GitHub API endpoint
```

Comment (ncphillips) [reply packages/react-tinacms-github/src/github-client/github-client.ts:335]: Yeah that looks good. `encoded` is also unnecessary right?

**Pull request merged**